spec/tree/builder/buildTreeSpec.js

```diff
@@ -0,0 +1,462 @@
+
+		const tree = buildTree( input );
+
+		const paragraph1 = new Paragraph( "" );
+		paragraph1.startIndex = 8;
+		paragraph1.endIndex = 14;
+		paragraph1.text = "Coffee";
+
+		const listItem1 = new ListItem();
+		listItem1.startIndex = 4;
+		listItem1.endIndex = 19;
+		listItem1.children = [ paragraph1 ];
+
+		const paragraph2 = new Paragraph( "" );
+		paragraph2.startIndex = 32;
+		paragraph2.endIndex = 35;
+		paragraph2.text = "Tea";
+
+		const structuredNode = new StructuredNode( "section" );
+		structuredNode.startIndex = 23;
+		structuredNode.endIndex = 45;
+		structuredNode.children = [ paragraph2 ];
+
+		const listItem2 = new ListItem();
+		listItem2.startIndex = 19;
+		listItem2.endIndex = 50;
+		listItem2.children = [ structuredNode ];
+
+		const list = new List( false );
+		list.startIndex = 0;
+		list.endIndex = 55;
+		list.children = [ listItem1, listItem2 ];
+
+		const expected = new StructuredNode( "root" );
+		expected.startIndex = 0;
+		expected.endIndex = 55;
+		expected.children = [ list ];
+
+		expect(  tree.toString() ).toEqual( expected.toString() );
+	} );
+
+
+	it( "can parse an HTML text into a StructuredNode with embedded children", () => {
+		const input = "<section><div>This sentence. Another sentence.</div></section>";
+
+		const tree = buildTree( input );
+
+		const paragraph = new Paragraph( "" );
+		paragraph.startIndex = 14;
+		paragraph.endIndex = 46;
+		paragraph.text = "This sentence. Another sentence.";
+
+		const structuredNode = new StructuredNode( "div" );
+		structuredNode.startIndex = 9;
+		structuredNode.endIndex = 52;
+		structuredNode.children = [ paragraph ];
+
+		const section = new StructuredNode( "section" );
+		section.startIndex = 0;
+		section.endIndex = 62;
+		section.children = [ structuredNode ];
+
+		const expected = new StructuredNode( "root" );
+		expected.startIndex = 0;
+		expected.endIndex = 62;
+		expected.children = [ section ];
+
+		expect(  tree.toString() ).toEqual( expected.toString() );
+	} );
+
+	it( "can parse an HTML text into a StructuredNode with a few siblings", () => {
+		const input = "<section><h1>First heading</h1><p>This sentence. Another sentence.</p></section>";
+
+		const tree = buildTree( input );
+
+		const heading = new Heading( 1 );
+		heading.startIndex = 9;
+		heading.endIndex = 31;
+		heading.text = "First heading";
+
+		const paragraph = new Paragraph( "p" );
+		paragraph.startIndex = 31;
+		paragraph.endIndex = 70;
+		paragraph.text = "This sentence. Another sentence.";
+
+		const section = new StructuredNode( "section" );
+		section.startIndex = 0;
+		section.endIndex = 80;
+		section.children = [ heading, paragraph ];
+
+		const expected = new StructuredNode( "root" );
+		expected.startIndex = 0;
+		expected.endIndex = 80;
+		expected.children = [ section ];
+
+		expect(  tree.toString() ).toEqual( expected.toString() );
+	} );
+
+	it( "can parse an irrelevant HTML-element and its contents into a StructuredIrrelevant node.", () => {
```

Comment:
```suggestion
	it( "can parse an irrelevant HTML element and its contents into a StructuredIrrelevant node.", () => {
```

Reply:
Spelling is ~hart~ hard 😛 . Going to fix it!